src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,913 @@
+2*≢1 {0#} ()
+
+suc≗1+ :  suc ≗ (1B +_)
+suc≗1+ 0#        =  refl
+suc≗1+ (2suc _)  =  refl
+suc≗1+ (suc2* _) =  refl
+
+suc-+ :  ∀ x y → (suc x) + y ≡ suc (x + y)
+suc-+ 0# y = begin
+  suc 0# + y     ≡⟨⟩
+  1B + y         ≡⟨ sym (suc≗1+ y) ⟩
+  suc y          ≡⟨⟩
+  suc (0# + y)   ∎
+
+suc-+ x 0# = begin
+  suc x + 0#     ≡⟨ +-identityʳ (suc x) ⟩
+  suc x          ≡⟨ cong suc (sym (+-identityʳ x)) ⟩
+  suc (x + 0#)   ∎
+
+suc-+ (2suc x) (2suc y) = begin
+  (suc (2suc x)) + (2suc y)    ≡⟨⟩
+  suc2* (suc x) + (2suc y)     ≡⟨⟩
+  suc (2suc (suc x + y))       ≡⟨ cong (suc ∘ 2suc) (suc-+ x y) ⟩
+  suc (2suc (suc (x + y)))     ≡⟨⟩
+  suc ((2suc x) + (2suc y))    ∎
+
+suc-+ (2suc x) (suc2* y) = begin
+  (suc (2suc x)) + (suc2* y)     ≡⟨⟩
+  (suc2* (suc x)) + (suc2* y)    ≡⟨⟩
+  suc (suc2* (suc x + y))        ≡⟨ cong (suc ∘ suc2*) (suc-+ x y) ⟩
+  suc (suc2* (suc (x + y)))      ≡⟨⟩
+  suc (suc (2suc (x + y)))       ≡⟨⟩
+  suc ((2suc x) + (suc2* y))     ∎
+
+suc-+ (suc2* x) (2suc y)  =  refl
+suc-+ (suc2* x) (suc2* y) =  refl
+
+
+1+≗suc :  (1B +_) ≗ suc
+1+≗suc = suc-+ 0#
+
+
+------------------------------------------------------------------------------
+-- Multiplication properties. Chapter I.
+------------------------------------------------------------------------------
+
+*-zeroˡ : LeftZero 0# _*_
+*-zeroˡ _ = refl
+
+*-zeroʳ : RightZero 0# _*_
+*-zeroʳ 0#        = refl
+*-zeroʳ (2suc _)  = refl
+*-zeroʳ (suc2* _) = refl
+
+*-zero : Zero 0# _*_
+*-zero = (*-zeroˡ , *-zeroʳ )
+
+x*y≡0⇒x≡0∨y≡0 :  ∀ x {y} → x * y ≡ 0# → x ≡ 0# ⊎ y ≡ 0#
+x*y≡0⇒x≡0∨y≡0 0# {_}  _ =  inj₁ refl
+x*y≡0⇒x≡0∨y≡0 _  {0#} _ =  inj₂ refl
+
+nz*nz :  ∀ {x y} → x ≢ 0# → y ≢ 0# → x * y ≢ 0#
+nz*nz {x} {_} x≢0 y≢0 xy≡0  with x*y≡0⇒x≡0∨y≡0 x xy≡0
+... | inj₁ x≡0 =  x≢0 x≡0
+... | inj₂ y≡0 =  y≢0 y≡0
+
+
+------------------------------------------------------------------------------
+-- Properties of toℕ and fromℕ. Chapter I.
+------------------------------------------------------------------------------
+
+toℕ∘2* :  ∀ x → toℕ (2* x) ≡ 2 *ₙ (toℕ x)
+toℕ∘2* 0#       = refl
+toℕ∘2* (2suc x) = begin
+  toℕ (2* (2suc x))         ≡⟨⟩
+  2 *ₙ (1+ (1+ (2 *ₙ m)))   ≡⟨⟩
+  2 *ₙ (2 +ₙ 2m)            ≡⟨ cong (2 *ₙ_) (sym (ℕp.*-distribˡ-+ 2 1 m)) ⟩
+  2 *ₙ (2 *ₙ (1+ m))        ≡⟨⟩
+  2 *ₙ (toℕ (2suc x))       ∎
+  where
+  m = toℕ x;  2m = 2 *ₙ m;  open P.≡-Reasoning
+
+toℕ∘2* (suc2* x) =  cong ((2 *ₙ_) ∘ 1+_) (toℕ∘2* x)
+
+
+toℕ∘suc :  ∀ x → toℕ (suc x) ≡ 1+ (toℕ x)
+toℕ∘suc 0#        =  refl
+toℕ∘suc (2suc x)  =  cong (1+_ ∘ (2 *ₙ_)) (toℕ∘suc x)
+toℕ∘suc (suc2* x) =  ℕp.*-distribˡ-+ 2 1 (toℕ x)
+
+toℕ∘pred :  ∀ x → toℕ (pred x) ≡ predₙ (toℕ x)
+toℕ∘pred 0#        =  refl
+toℕ∘pred (2suc x)  =  cong predₙ $ sym $ ℕp.*-distribˡ-+ 2 1 (toℕ x)
+toℕ∘pred (suc2* x) =  toℕ∘2* x
+
+
+x≡0⇒toℕ-x≡0 :  ∀ {x} → x ≡ 0# → toℕ x ≡ 0
+x≡0⇒toℕ-x≡0 {0#} _ = refl
+
+toℕ-x≡0⇒x≡0 :  ∀ {x} → toℕ x ≡ 0 → x ≡ 0#
+toℕ-x≡0⇒x≡0 {0#} _ = refl
+
+toℕ+homo :  ∀ x y → toℕ (x + y) ≡ toℕ x +ₙ toℕ y
```

Comment:
The names of these homomorphism proofs should follow the pattern `toℕ-homo-+`

Reply:
Done for ``+`` and ``*``.